diff --git a/node_modules/@equilibria/root/accumulator/types/Accumulator6.sol b/node_modules/@equilibria/root/accumulator/types/Accumulator6.sol
index 6faf409..e993b7a 100644
--- a/node_modules/@equilibria/root/accumulator/types/Accumulator6.sol
+++ b/node_modules/@equilibria/root/accumulator/types/Accumulator6.sol
@@ -43,6 +43,7 @@ library Accumulator6Lib {
      * @param total Denominator of the ratio
      */
     function increment(Accumulator6 memory self, Fixed6 amount, UFixed6 total) internal pure {
+        if (amount.isZero()) return;
         self._value = self._value.add(_div(amount, total));
     }
 
@@ -54,6 +55,7 @@ library Accumulator6Lib {
      * @param total Denominator of the ratio
      */
     function decrement(Accumulator6 memory self, Fixed6 amount, UFixed6 total) internal pure {
+        if (amount.isZero()) return;
         self._value = self._value.add(_div(amount.mul(Fixed6Lib.NEG_ONE), total));
     }
 
diff --git a/node_modules/@equilibria/root/accumulator/types/UAccumulator6.sol b/node_modules/@equilibria/root/accumulator/types/UAccumulator6.sol
index e2117c0..778b2e8 100644
--- a/node_modules/@equilibria/root/accumulator/types/UAccumulator6.sol
+++ b/node_modules/@equilibria/root/accumulator/types/UAccumulator6.sol
@@ -41,6 +41,7 @@ library UAccumulator6Lib {
      * @param total Denominator of the ratio
      */
     function increment(UAccumulator6 memory self, UFixed6 amount, UFixed6 total) internal pure {
+        if (amount.isZero()) return;
         self._value = self._value.add(amount.div(total));
     }
 }
