diff --git a/node_modules/@perennial/sdk/dist/lib/oracle/index.js b/node_modules/@perennial/sdk/dist/lib/oracle/index.js
index e220dc4..e305a6c 100644
--- a/node_modules/@perennial/sdk/dist/lib/oracle/index.js
+++ b/node_modules/@perennial/sdk/dist/lib/oracle/index.js
@@ -31,7 +31,7 @@ async function oracleCommitmentsLatest({ chainId, oracleClients, requests, versi
         }, new Map());
         // Generate commitment(s) gor each factory grouping
         const commitmentPromises = Array.from(groupedRequests.entries()).map(async ([factory, reqs]) => {
-            const providerType = await oracleProviderTypeForFactoryAddress({ publicClient, factory });
+            const providerType = reqs[0].factoryType ?? (await oracleProviderTypeForFactoryAddress({ publicClient, factory }));
             if (providerType === 'PythFactory') {
                 const pythResponse = await (0, pyth_1.buildCommitmentsForOracles)({
                     chainId,
@@ -79,7 +79,7 @@ async function oracleCommitmentsTimestamp({ chainId, oracleClients, requests, ti
         }, new Map());
         // Generate commitment(s) gor each factory grouping
         const commitmentPromises = Array.from(groupedRequests.entries()).map(async ([factory, reqs]) => {
-            const providerType = await oracleProviderTypeForFactoryAddress({ publicClient, factory });
+            const providerType = reqs[0].factoryType ?? (await oracleProviderTypeForFactoryAddress({ publicClient, factory }));
             if (providerType === 'PythFactory') {
                 const pythResponse = await (0, pyth_1.buildCommitmentsForOracles)({
                     chainId,
